include/private/dns/getaddrinfo_async.hpp

```diff
@@ -61,7 +61,11 @@ std::vector<Answer> getaddrinfo_async_parse_response(const std::string &name,
             throw GenericError(); /* Avoid g++ warning */
         }
         if (p->ai_canonname != nullptr) {
-            answer.hostname = p->ai_canonname;
+            Answer cname_ans = answer;
```

Comment:
Is this an API change? Was `answer.hostname` ever used? We should at the very least document it in the changelog.

Reply:
No, there was no use case where we were using this. I'd say it was unexpected behavior (i.e. inconsistent with our API) what we were doing previously.